### ISSUE TITLE
Unable to Reproduce Results with Provided YAML Config (FairGBM)

### ISSUE THREAD

**Issue by dattt19uit** [feedzai/fairgbm#50]
Hi, and thank you for releasing FairGBM.

We are currently working on a project related to gradient boosting models and attempted to reproduce the results from your paper using the arxiv-supp-materials branch and the provided YAML file. However, we encountered issues when running the pipeline, particularly when loading the YAML config for FairGBM and LightGBM. The performance metrics are also significantly worse than reported.

Our implementation (with the YAML file loaded) is available here for your reference:
🔗 https://www.kaggle.com/code/haohoangofficial/fairgbm-cs2229-ch190?scriptVersionId=241619685

We would appreciate your help in understanding how to correctly apply the hyperparameters, especially how to properly load and apply the YAML configuration to reproduce the results for FairGBM and the baseline models.

Thanks in advance!

Best regards,
Dat Tran (on behalf of our student team)

**Comment by AndreFCruz** [comment]
Hi Dat,
This looks like it might be a mismatch of dependencies. Requirements are not pinned, and the lightgbm binaries are compiled to your machine so it may lead to issues on different platforms.
Preferably run FairGBM on the provided docker image to reproduce any results.

All the best
